packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -203,41 +203,54 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
 
   @autobind
   private _onDropdownKeyDown(ev: React.KeyboardEvent<HTMLElement>) {
+    let eventHandled: boolean = false;
+
     switch (ev.which) {
       case KeyCodes.enter:
         this.setState({
           isOpen: !this.state.isOpen
         });
+        eventHandled = true;
         break;
 
       case KeyCodes.escape:
-        this.setState({
-          isOpen: false
-        });
+        if (this.state.isOpen) {
+          this.setState({
+            isOpen: false
+          });
+          eventHandled = true;
+        }
```

Comment:
Right but those other cases are handled in the `default` case... which returns early.